packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/heatmap.dart

```diff
@@ -0,0 +1,245 @@
+    double? maxIntensityParam,
+    double? opacityParam,
+    int? radiusParam,
+    int? minimumZoomIntensityParam,
+    int? maximumZoomIntensityParam,
+  }) {
+    return Heatmap(
+      heatmapId: heatmapId,
+      data: dataParam ?? data,
+      dissipating: dissipatingParam ?? dissipating,
+      gradient: gradientParam ?? gradient,
+      maxIntensity: maxIntensityParam ?? maxIntensity,
+      opacity: opacityParam ?? opacity,
+      radius: radiusParam ?? radius,
+      minimumZoomIntensity: minimumZoomIntensityParam ?? minimumZoomIntensity,
+      maximumZoomIntensity: maximumZoomIntensityParam ?? maximumZoomIntensity,
+    );
+  }
+
+  /// Creates a new [Heatmap] object whose values are the same as this
+  /// instance.
+  @override
+  Heatmap clone() => copyWith(
+        dataParam: List<WeightedLatLng>.of(data),
+        gradientParam: gradient?.clone(),
+      );
+
+  /// Converts this object to something serializable in JSON.
+  @override
+  Object toJson() {
+    final Map<String, Object> json = <String, Object>{};
+
+    void addIfPresent(String fieldName, Object? value) {
+      if (value != null) {
+        json[fieldName] = value;
+      }
+    }
+
+    addIfPresent('heatmapId', heatmapId.value);
+    addIfPresent('data', data.map((WeightedLatLng e) => e.toJson()).toList());
+    addIfPresent('dissipating', dissipating);
+    addIfPresent('gradient', gradient?.toJson());
+    addIfPresent('maxIntensity', maxIntensity);
+    addIfPresent('opacity', opacity);
+    addIfPresent('radius', radius);
+    addIfPresent('minimumZoomIntensity', minimumZoomIntensity);
+    addIfPresent('maximumZoomIntensity', maximumZoomIntensity);
+
+    return json;
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) {
+      return true;
+    }
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
+    return other is Heatmap &&
+        heatmapId == other.heatmapId &&
+        listEquals(data, other.data) &&
+        dissipating == other.dissipating &&
+        gradient == other.gradient &&
+        maxIntensity == other.maxIntensity &&
+        opacity == other.opacity &&
+        radius == other.radius &&
+        minimumZoomIntensity == other.minimumZoomIntensity &&
+        maximumZoomIntensity == other.maximumZoomIntensity;
+  }
+
+  @override
+  int get hashCode => heatmapId.hashCode;
+}
+
+/// Represents a mapping of intensity to color.  Interpolates between given set
+/// intensity and color values to produce a full mapping for the range [0, 1].
+@immutable
+class HeatmapGradient {
+  /// Creates a new [HeatmapGradient] object.
+  const HeatmapGradient({
+    required this.colors,
+    required this.startPoints,
+    this.colorMapSize = 256,
+  })  : assert(colors.length == startPoints.length),
+        assert(colors.length > 0),
+        assert(startPoints.length > 0);
+
+  /// The specific colors for the specific intensities specified by startPoints.
```

Comment:
Nothing

Reply:
I assume web just interpolates between the colors in a uniform distribution

Reply:
We'll need to figure out what it does and document it; currently this description just doesn't make sense for a platform that doesn't support "startPoints", because in that case it effectively reads: "The specific colors for the specific intensities specified by nothing".

Reply:
Done